File: pkg/validate/validate_test.go
package validate

import (
	. "github.com/onsi/ginkgo"
	. "github.com/onsi/gomega"
)

var _ = Describe("Param", func() {
	Context("New()", func() {
		It("should not return nil", func() {
			/* arrange/act/assert */
			Expect(New()).Should(Not(BeNil()))
		})
	})
})

File: sdk/js/src/opspec/interpreter/opcall/params/param/boolean/validate.test.ts
import objectUnderTest from './validate'

describe('validate', () => {
  describe('typeof value isn\'t boolean', () => {
    it('returns expected result', () => {
      /* arrange */
      const expectedResult = [new Error('boolean required')]

      /* act */
      const actualResult = objectUnderTest(
        undefined as unknown as boolean
      )

      /* assert */
      expect(actualResult).toEqual(expectedResult)
    })
  })
  describe('typeof value is boolean', () => {
    it('returns expected result', () => {
      /* arrange/act */
      const actualResult = objectUnderTest(true)

      /* assert */
      expect(actualResult).toEqual([])
    })
  })
})
